nni/retiarii/nn/pytorch/mutator.py

```diff
@@ -144,14 +144,13 @@ def _retrieve_chain_from_graph(self, graph: Graph) -> List[Node]:
         return chain
 
     def mutate(self, model):
-        min_depth = self.nodes[0].operation.parameters['min_depth']
-        max_depth = self.nodes[0].operation.parameters['max_depth']
-        if min_depth < max_depth:
-            chosen_depth = self.choice(list(range(min_depth, max_depth + 1)))
         for node in self.nodes:
             # the logic here is similar to layer choice. We find cell attached to each node.
             target: Graph = model.graphs[node.operation.cell_name]
             chain = self._retrieve_chain_from_graph(target)
+            # and we get the chosen depth (by value choice)
+            node_in_model = model.get_node_by_name(node.name)
+            chosen_depth = node_in_model.operation.parameters['depth']
```

Comment:
what is the type of `chosen_depth`? int or valuechoice?

Reply:
which part decides the chosen depth?

Reply:
It's originally a ValueChoice, but it has already been mutated by a ParameterChoiceMutator here.